### PR TITLE
Add caching to ACL

### PR DESCRIPTION
**Pull request by RubenVerborgh**
Avoid that the same .acl files are read and parsed multiple times per request.

### CODE REVIEW

**Comment by RubenVerborgh** [comment]
@dmitrizagidulin You might want to have this as a merge commit rather than rebase, in order to keep the commits together.

**Comment by dmitrizagidulin** [comment]
@RubenVerborgh good call, re merge

File: lib/acl-checker.js [hunk 1]
```diff
@@ -1,172 +1,143 @@
...
+
+    // Obtain the permission set for the resource
+    if (!this._permissionSet) {
+      this._permissionSet = this.getNearestACL()
+        .then(acl => this.getPermissionSet(acl))
+    }
+
+    // Check the resource's permissions
+    return this._permissionSet.then(acls => this.checkAccess(acls, user, mode))
+    .catch(err => {
```
Comment (dmitrizagidulin) [inline lib/acl-checker.js]: I think this should be indented? Something like:

```js
return this._permissionSet
  .then(acls => this.checkAccess(acls, user, mode))
  .catch(err => {
  })
```
?

Comment (RubenVerborgh) [inline lib/acl-checker.js]: Makes sense, will do. (`standard` didn't complain; but I'd be the last person to say that it is a good style checker :wink:)

Comment (RubenVerborgh) [inline lib/acl-checker.js]: Done in bda742636e346fff02f6f082b10d37bbd524013b, let me know if more indentation needs adjustments.

**Comment by dmitrizagidulin** [comment]
Oh man, great work on this!

**Pull request merged**